includes/Traits/Post_Type.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Trait Post_Type
+ *
+ * @package   Google\Web_Stories\Traits
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Traits;
+
+use WP_Post_Type;
+use WP_REST_Controller;
+use WP_REST_Posts_Controller;
+
+/**
+ * Trait Post_Type
+ *
+ * @package Google\Web_Stories\Traits
+ */
+trait Post_Type {
+	/**
+	 * Get rest base name based on the post type slug.
+	 *
+	 * @since 1.7.0
+	 *
+	 * @param string $slug The post type slug.
+	 *
+	 * @return string Rest base.
+	 */
+	protected function get_post_type_rest_base( $slug ) {
```

Comment:
Renaming `$slug` to `$post_type` everywhere would make it more clear that we're dealing with post type slugs and not post permalink slugs.